pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statesinformer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+)
+
+func (s *statesInformer) syncNodeResourceTopology(node *corev1.Node) {
+	topologyName := fmt.Sprintf("koord-%s", node.Name)
+	ctx := context.TODO()
+	blocker := true
+	_, err := s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Get(ctx, topologyName, metav1.GetOptions{ResourceVersion: "0"})
+	if err == nil {
+		return
+	}
+	if !errors.IsNotFound(err) {
+		klog.Errorf("failed to get node resource topology %s, err: %v", topologyName, err)
+		return
+	}
+
+	topology := &v1alpha1.NodeResourceTopology{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: topologyName,
+			Labels: map[string]string{
+				extension.LabelManagedBy: "Koordinator",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "v1",
+					Kind:               "Node",
+					Name:               node.Name,
+					UID:                node.GetUID(),
+					Controller:         &blocker,
+					BlockOwnerDeletion: &blocker,
+				},
+			},
+		},
+		// fields are required
+		TopologyPolicies: []string{"None"},
+		Zones:            v1alpha1.ZoneList{v1alpha1.Zone{Name: "fake-name", Type: "fake-type"}},
+	}
+	_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Create(ctx, topology, metav1.CreateOptions{})
+	if err != nil {
+		klog.Errorf("failed to create node resource topology %s, err: %v", topologyName, err)
+		return
+	}
+
+}
+
+func (s *statesInformer) reportNodeTopology() {
```

Comment:
better to divide this function into three functions:
1. generate cpuTopo
2. calcu cpu share pool
3. update crd